connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -350,54 +437,49 @@ public Task newTask(Class<? extends Task> taskClass) {
      * @throws ConnectException if the {@link Converter} implementation class could not be found
      */
     public Converter newConverter(AbstractConfig config, String classPropertyName, ClassLoaderUsage classLoaderUsage) {
+        return getConverter(config, classPropertyName, null, classLoaderUsage);
+    }
+
+    /**
+     * Used to get a versioned converter. It will always try and get the converter from the set of plugin classloaders.
+     *
+     * @param config              the configuration containing the {@link Converter}'s configuration; may not be null
+     * @param classPropertyName   the name of the property that contains the name of the {@link Converter} class; may not be null
+     * @param versionPropertyName the name of the property that contains the version of the {@link Converter} class; may not be null
+     * @return the instantiated and configured {@link Converter}; null if the configuration did not define the specified property
+     * @throws ConnectException if the {@link Converter} implementation class could not be found,
+     * @throws VersionedPluginLoadingException if the version requested is not found
+     */
+    public Converter newConverter(AbstractConfig config, String classPropertyName, String versionPropertyName) {
+        ClassLoaderUsage classLoader = config.getString(versionPropertyName) == null ? ClassLoaderUsage.CURRENT_CLASSLOADER: ClassLoaderUsage.PLUGINS;
+        return getConverter(config, classPropertyName, versionPropertyName, classLoader);
+    }
+
+    private Converter getConverter(AbstractConfig config, String classPropertyName, String versionPropertyName, ClassLoaderUsage classLoaderUsage) {
         if (!config.originals().containsKey(classPropertyName)) {
             // This configuration does not define the converter via the specified property name
             return null;
         }
-
```

Comment:
I think it was a good idea to deduplicate this switch-case. There's another copy in newConfigProvider that we can get rid of too, and just have it use a `null` `VersionRange`.